system/Config/BaseConfig.php

```diff
@@ -96,13 +96,39 @@ protected function initEnvValue(&$property, string $name, string $prefix, string
             foreach (array_keys($property) as $key) {
                 $this->initEnvValue($property[$key], "{$name}.{$key}", $prefix, $shortPrefix);
             }
-        } elseif (($value = $this->getEnvValue($name, $prefix, $shortPrefix)) !== false && $value !== null) {
-            if ($value === 'false') {
-                $value = false;
-            } elseif ($value === 'true') {
-                $value = true;
+        } else {
+            $value = $this->getEnvValue($name, $prefix, $shortPrefix);
+
+            if ($value !== null) {
+                $lowered = strtolower($value);
+
+                switch (true) {
+                    case $lowered === 'false':
+                        $value = false;
+                        break;
+
+                    case $lowered === 'true':
+                        $value = true;
+                        break;
+
+                    case $lowered === 'empty':
+                        $value = '';
+                        break;
+
+                    case $lowered === 'null':
+                        $value = null;
+                        break;
+
+                    default:
+                        $value = trim($value, '\'"');
+                }
+
+                if (is_numeric($value)) {
+                    $value = strpos($value, '.') !== false ? (float) $value : (int) $value;
```

Comment:
I think this is accurate but can't test on mobile... `is_numeric()` accounts for multiple periods, right?
```php
is_numeric('4.1.8') // false
```

Reply:
The check returns false.
https://3v4l.org/nTQpm